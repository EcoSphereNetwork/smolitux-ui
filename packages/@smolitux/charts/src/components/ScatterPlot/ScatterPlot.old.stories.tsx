import type { Meta, StoryObj } from '@storybook/react';
import { ScatterPlot.old } from './ScatterPlot.old';

const meta: Meta<typeof ScatterPlot.old> = {
  title: 'Components/ScatterPlot.old',
  component: ScatterPlot.old,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'ScatterPlot.old',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom ScatterPlot.old',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive ScatterPlot.old',
    onClick: () => alert('Clicked!'),
  },
};
