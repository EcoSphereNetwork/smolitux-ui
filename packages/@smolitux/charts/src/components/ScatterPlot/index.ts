// packages/@smolitux/charts/src/components/ScatterPlot/index.ts
export { default, type ScatterPlotProps, type ScatterPlotSeries, type ScatterPlotDataPoint } from './ScatterPlot';
