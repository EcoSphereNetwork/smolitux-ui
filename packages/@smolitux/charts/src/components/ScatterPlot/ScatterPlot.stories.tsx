import type { Meta, StoryObj } from '@storybook/react';
import { ScatterPlot } from './ScatterPlot';

const meta: Meta<typeof ScatterPlot> = {
  title: 'Components/ScatterPlot',
  component: ScatterPlot,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'ScatterPlot',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom ScatterPlot',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive ScatterPlot',
    onClick: () => alert('Clicked!'),
  },
};
