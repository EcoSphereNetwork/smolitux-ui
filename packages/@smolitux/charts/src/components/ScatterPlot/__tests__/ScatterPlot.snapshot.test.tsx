import React from 'react';
import { render } from '@testing-library/react';
import { ScatterPlot } from '../ScatterPlot';

jest.mock('@smolitux/theme', () => ({
  useTheme: () => ({ themeMode: 'light' }),
}));

describe('ScatterPlot Snapshots', () => {
  const mockData = {
    id: 'dataset1',
    name: 'Dataset 1',
    data: [
      { x: 10, y: 20, size: 5 },
      { x: 15, y: 30, size: 8 },
      { x: 20, y: 15, size: 6 },
      { x: 25, y: 40, size: 10 },
      { x: 30, y: 25, size: 7 },
      { x: 35, y: 35, size: 9 },
    ],
  };

  it('renders default chart', () => {
    const { asFragment } = render(<ScatterPlot data={mockData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
