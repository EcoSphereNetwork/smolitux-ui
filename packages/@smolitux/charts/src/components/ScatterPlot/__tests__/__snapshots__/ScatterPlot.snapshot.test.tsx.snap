// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScatterPlot Snapshots renders default chart 1`] = `
<DocumentFragment>
  <svg
    class="w-full "
    height="300"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 1000 563"
    width="100%"
  >
    <rect
      fill="#F9FAFB"
      height="230"
      stroke="#E5E7EB"
      stroke-width="1"
      width="100%"
      x="50"
      y="30"
    />
    <g
      class="grid-lines"
    >
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="50"
        x2="NaN"
        y1="260"
        y2="260"
      />
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="50"
        x2="NaN"
        y1="202.5"
        y2="202.5"
      />
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="50"
        x2="NaN"
        y1="145"
        y2="145"
      />
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="50"
        x2="NaN"
        y1="87.5"
        y2="87.5"
      />
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="50"
        x2="NaN"
        y1="30"
        y2="30"
      />
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="NaN"
        x2="NaN"
        y1="30"
        y2="260"
      />
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="NaN"
        x2="NaN"
        y1="30"
        y2="260"
      />
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="NaN"
        x2="NaN"
        y1="30"
        y2="260"
      />
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="NaN"
        x2="NaN"
        y1="30"
        y2="260"
      />
      <line
        stroke="#E5E7EB"
        stroke-dasharray="4,4"
        stroke-width="1"
        x1="NaN"
        x2="NaN"
        y1="30"
        y2="260"
      />
    </g>
    <g
      class="data-points"
    >
      <g
        class="series"
      >
        <circle
          class="transition-opacity animate-fade-in"
          cursor="default"
          cx="NaN"
          cy="207.72727272727272"
          fill="#3B82F6"
          r="2.5"
          stroke="rgba(255,255,255,0.5)"
          stroke-width="1"
          style="opacity: 0; animation: fadeIn 0.5s ease-out forwards;"
        />
        <circle
          class="transition-opacity animate-fade-in"
          cursor="default"
          cx="NaN"
          cy="124.09090909090908"
          fill="#3B82F6"
          r="4"
          stroke="rgba(255,255,255,0.5)"
          stroke-width="1"
          style="opacity: 0; animation: fadeIn 0.5s ease-out forwards;"
        />
        <circle
          class="transition-opacity animate-fade-in"
          cursor="default"
          cx="NaN"
          cy="249.54545454545456"
          fill="#3B82F6"
          r="3"
          stroke="rgba(255,255,255,0.5)"
          stroke-width="1"
          style="opacity: 0; animation: fadeIn 0.5s ease-out forwards;"
        />
        <circle
          class="transition-opacity animate-fade-in"
          cursor="default"
          cx="NaN"
          cy="40.454545454545446"
          fill="#3B82F6"
          r="5"
          stroke="rgba(255,255,255,0.5)"
          stroke-width="1"
          style="opacity: 0; animation: fadeIn 0.5s ease-out forwards;"
        />
        <circle
          class="transition-opacity animate-fade-in"
          cursor="default"
          cx="NaN"
          cy="165.90909090909088"
          fill="#3B82F6"
          r="3.5"
          stroke="rgba(255,255,255,0.5)"
          stroke-width="1"
          style="opacity: 0; animation: fadeIn 0.5s ease-out forwards;"
        />
        <circle
          class="transition-opacity animate-fade-in"
          cursor="default"
          cx="NaN"
          cy="82.27272727272728"
          fill="#3B82F6"
          r="4.5"
          stroke="rgba(255,255,255,0.5)"
          stroke-width="1"
          style="opacity: 0; animation: fadeIn 0.5s ease-out forwards;"
        />
      </g>
    </g>
    <g
      class="x-axis"
    >
      <line
        stroke="#9CA3AF"
        stroke-width="1"
        x1="50"
        x2="NaN"
        y1="260"
        y2="260"
      />
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="NaN"
          x2="NaN"
          y1="260"
          y2="265"
        />
        <text
          fill="#4B5563"
          font-size="12"
          text-anchor="middle"
          x="NaN"
          y="280"
        >
          8.75
        </text>
      </g>
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="NaN"
          x2="NaN"
          y1="260"
          y2="265"
        />
        <text
          fill="#4B5563"
          font-size="12"
          text-anchor="middle"
          x="NaN"
          y="280"
        >
          15.625
        </text>
      </g>
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="NaN"
          x2="NaN"
          y1="260"
          y2="265"
        />
        <text
          fill="#4B5563"
          font-size="12"
          text-anchor="middle"
          x="NaN"
          y="280"
        >
          22.5
        </text>
      </g>
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="NaN"
          x2="NaN"
          y1="260"
          y2="265"
        />
        <text
          fill="#4B5563"
          font-size="12"
          text-anchor="middle"
          x="NaN"
          y="280"
        >
          29.375
        </text>
      </g>
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="NaN"
          x2="NaN"
          y1="260"
          y2="265"
        />
        <text
          fill="#4B5563"
          font-size="12"
          text-anchor="middle"
          x="NaN"
          y="280"
        >
          36.25
        </text>
      </g>
    </g>
    <g
      class="y-axis"
    >
      <line
        stroke="#9CA3AF"
        stroke-width="1"
        x1="50"
        x2="50"
        y1="30"
        y2="260"
      />
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="45"
          x2="50"
          y1="260"
          y2="260"
        />
        <text
          dominant-baseline="middle"
          fill="#4B5563"
          font-size="12"
          text-anchor="end"
          x="40"
          y="260"
        >
          13.75
        </text>
      </g>
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="45"
          x2="50"
          y1="202.5"
          y2="202.5"
        />
        <text
          dominant-baseline="middle"
          fill="#4B5563"
          font-size="12"
          text-anchor="end"
          x="40"
          y="202.5"
        >
          20.625
        </text>
      </g>
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="45"
          x2="50"
          y1="145"
          y2="145"
        />
        <text
          dominant-baseline="middle"
          fill="#4B5563"
          font-size="12"
          text-anchor="end"
          x="40"
          y="145"
        >
          27.5
        </text>
      </g>
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="45"
          x2="50"
          y1="87.5"
          y2="87.5"
        />
        <text
          dominant-baseline="middle"
          fill="#4B5563"
          font-size="12"
          text-anchor="end"
          x="40"
          y="87.5"
        >
          34.375
        </text>
      </g>
      <g>
        <line
          stroke="#9CA3AF"
          stroke-width="1"
          x1="45"
          x2="50"
          y1="30"
          y2="30"
        />
        <text
          dominant-baseline="middle"
          fill="#4B5563"
          font-size="12"
          text-anchor="end"
          x="40"
          y="30"
        >
          41.25
        </text>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;
